### PR TITLE
feat(@desktop/wallet): Implement loading animation

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
### What does the PR do

Implements generic loading state support for:
1. StatusBaseText
2. StatusLIstItem
3. StatusSmartIdenticon
as per new designs https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=9476%3A104890&t=PI0EIQ2jwLFNhOxf-4

### Affected areas
1. StatusBaseText
2. StatusLIstItem
3. StatusSmartIdenticon

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://user-images.githubusercontent.com/60327365/210858286-ef82bd4b-30bc-411c-bed0-5eeaf4c23579.mov

### CODE REVIEW

**Comment by Khushboo-dev-cpp** [comment]
@stefandunca @dlipicar @alaibe this is still a draft and I still need to clean up the code, but have a look at the video and the code. After taking into consideration into points from you guys, I think this is the best way forward. 

1. Because we dont want to maintain two pieces of code that are meant to look similar but dont because say the the StatusLIstItem design changes. We will need to update any design changes in both the "actual" and "loading" version of the item. 
2. Also considering that this loading state is meant to be generic all throughout our app, this approach seems beneficial to us.

Let me know your thoughts and we can discuss it in tomorrow stand up as well

**Comment by Khushboo-dev-cpp** [comment]
@alexandraB99 ! we actually need help from the UI team to let us know if this approach is something we agree with. Please check the design links I have shared in the ticket and the video attached.

Even though this is only currently showing up for wallet, it was discussed that will be the approach to loading states in application eventually which is why I think embedding it with the basic components is a good idea. Let me know :)

File: ui/StatusQ/src/StatusQ/Components/LoadingComponent.qml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import QtQuick 2.13
+
+import StatusQ.Core.Theme 0.1
+import QtGraphicalEffects 1.12
+import StatusQ.Animations 0.1
+
+Rectangle {
+    id: root
+    property bool loading
```
Comment (alexandraB99) [inline ui/StatusQ/src/StatusQ/Components/LoadingComponent.qml:9]: could you please make this a `Controls` with `background` and `contentItem` properties set instead?

File: ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        active: root.loading
+        sourceComponent: LoadingComponent {
+            anchors.centerIn: parent
+            radius: textMetrics.font.pixelSize === 15 ? 4 : 8
+            height: textMetrics.tightBoundingRect.height
+            width: Math.min(root.width, 140)
+            loading: root.loading
+        }
+    }
+    TextMetrics {
```
Comment (alexandraB99) [inline ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:24]: is this used somewhere?

File: ui/StatusQ/src/StatusQ/Core/Theme/StatusDarkTheme.qml [hunk 3]
```diff
@@ -62,6 +62,9 @@ ThemePalette {
 
     statusFloatingButtonHighlight: getColor('blue4', 0.3)
 
+//    statusLoadingHighlight: getColor('white', 0.07)
+//    statusLoadingHighlight2: getColor('lightPattensBlue', 0.03)
+
```
Comment (alexandraB99) [inline ui/StatusQ/src/StatusQ/Core/Theme/StatusDarkTheme.qml:67]: please remove if not needed/used

File: ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+StatusBaseText {
+    id: root
+    property bool loading: false
+    property color customColor: Theme.palette.directColor1
+    color: loading ? "transparent" : customColor
+    Loader {
+        anchors.left: parent.left
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:13]: It's better to set both anchors here

**Review by caybro — COMMENTED** [review]
Overall nice work!

Please make it a proper `Control` as Alex mentioned; bonus points for putting this into Storybook ;)

File: ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:24]: yes I need it for getting the height of the letter

File: ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml (hunk 4 above)
Comment (Khushboo-dev-cpp) [reply ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:13]: wdym?

Comment (caybro) [reply ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:13]: Also `anchors.right`

Comment (Khushboo-dev-cpp) [reply ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:47]: wont be needed as I specified a width in the loaders source component

Comment (caybro) [reply ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:47]: OK

**Comment by alexandraB99** [comment]
> Overall nice work!
> 
> Please make it a proper `Control` as Alex mentioned; bonus points for putting this into Storybook ;)

@caybro good point, should we just not add it to sandbox and just add it in storybook instead? wdyt?

**Comment by Khushboo-dev-cpp** [comment]
@alexandraB99 @caybro addresses all the comments for the bonus points for storybook :D

let me know if we can proceed with this task, also I am guessing we will be redoing the StatusQ components after MVP and new design system maybe we can move all of the demos from sandbox to story book then?

Write now its all here and there, I am also adding the loading state to StatusLIstItem and for that I need to use sandbox as I dont thats migrated to storybook yet?

**Review by caybro — APPROVED** [review]

**Review by alaibe — APPROVED** [review]

**Pull request merged**